File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        m,n,cnt = len(grid), len(grid[0]), 0
        r, c = m-1, 0
        while r >= 0 and c < n:
            if grid[r][c] < 0:
                cnt += n - c
                r -= 1
            else:
                c += 1
        return cnt

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int m = grid.size(), n= grid[0].size(), r = m-1, c= 0, cnt = 0;
        while(r>=0 and c<n){
            if(grid[r][c]<0){
                --r;
                cnt += n-c;
            }
            else
                c++;
        }
        return cnt;
    }
};